src/Controller/CaseController.php

```diff
@@ -356,6 +358,18 @@ public function rescheduleFinishHearingDeadline(CaseEntity $case, Request $reque
         $rescheduleForm->handleRequest($request);
 
         if ($rescheduleForm->isSubmitted() && $rescheduleForm->isValid()) {
+            $case->setHasReachedHearingDeadline(false);
+
+            // Make sure to modify process deadline if new hearing deadline occurs later than current process deadline
+            if ($case->getFinishHearingDeadline() >= $case->getFinishProcessingDeadline()) {
+                $newHearingDeadline = $case->getFinishHearingDeadline()->format('d/m/Y');
+
+                $newProcessingDeadline = \DateTime::createFromFormat('d/m/Y', $newHearingDeadline)->modify('+1 day');
```

Comment:
If I understand this correctly, you want to get a date 1 day after `$case->getFinishHearingDeadline()`.
Rather than converting to and back from a string (which works), you should clone the original date and then modify the clone, i.e.

```php
$newProcessingDeadline = clone $case->getFinishHearingDeadline();
$newProcessingDeadline->modify('+1 day');
```

Reply:
Correct, added some comments to make this clearer and used cloning method.